plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelsAction.java

```diff
@@ -157,6 +171,94 @@ private void undeployModels(String[] targetNodeIds, String[] modelIds, ActionLis
         MLUndeployModelNodesRequest mlUndeployModelNodesRequest = new MLUndeployModelNodesRequest(targetNodeIds, modelIds);
 
         client.execute(MLUndeployModelAction.INSTANCE, mlUndeployModelNodesRequest, ActionListener.wrap(r -> {
+            if (r.getNodes() != null) {
```

Comment:
We are moving big pieces of codes here and there but it's not affecting any unit tests, that makes me scared, lol